ENIGMAsystem/SHELL/Audio_Systems/DirectSound/SoundResource.h

```diff
@@ -36,15 +36,19 @@ struct SoundResource {
   bool idle;          // True if this sound is not being used, false if playing or paused.
   bool playing;       // True if this sound is playing; not paused or idle.
 
-  SoundResource() : soundBuffer(0), cleanup(0), userdata(0), seek(0), type(0), kind(0),
+  Sound() : soundBuffer(0), cleanup(0), userdata(0), seek(0), type(0), kind(0),
     loaded(LOADSTATE_NONE), idle(1), playing(0) {}
 
-  ~SoundResource() {
+  void destroy() {
```

Comment:
You should probably call `destroy()` in your own destructor. Otherwise, this class is not generally-usable, and should only be visible to AssetArray. Also, that 0 is nullptr, yes?

Reply:
Yes, that 0 is nullptr, but why would I actually type that out.

Reply:
Ok: 42e7991af85d5f3c2102475f5685a461e1fa7e18

I very nicely gave you a destructor for Sound.